server/embed/config_test.go

```diff
@@ -92,136 +92,99 @@ func TestConfigFileOtherFields(t *testing.T) {
-			experimentalStopGRPCServiceOnDefrag: "true",
-			expectedFeatures: map[featuregate.Feature]bool{
-				features.DistributedTracing:      true,
-				features.StopGRPCServiceOnDefrag: true,
-				features.InitialCorruptCheck:     false,
-			},
+			name:              "can set feature gate to false from experimental flag",
+			experimentalKey:   experimentalKey,
+			experimentalValue: false,
+			expectedFeatures:  defaultFeatureGatesWithFeaturegate(fg, false),
 		},
 		{
-			name:                                "can set feature gate to true from experimental flag",
-			experimentalStopGRPCServiceOnDefrag: "true",
-			expectedFeatures: map[featuregate.Feature]bool{
-				features.StopGRPCServiceOnDefrag: true,
-				features.DistributedTracing:      false,
-				features.InitialCorruptCheck:     false,
-			},
+			name:                   "can set feature gate " + fgname + " to true from feature gate flag",
+			serverFeatureGatesJSON: fgname + "=true",
+			expectedFeatures:       defaultFeatureGatesWithFeaturegate(fg, true),
 		},
 		{
-			name:                                "can set feature gate to false from experimental flag",
-			experimentalStopGRPCServiceOnDefrag: "false",
-			expectedFeatures: map[featuregate.Feature]bool{
-				features.StopGRPCServiceOnDefrag: false,
-				features.DistributedTracing:      false,
-				features.InitialCorruptCheck:     false,
-			},
-		},
-		{
-			name:                            "can set feature gate experimentalInitialCorruptCheck to true from experimental flag",
-			experimentalInitialCorruptCheck: "true",
-			expectedFeatures: map[featuregate.Feature]bool{
-				features.StopGRPCServiceOnDefrag: false,
-				features.DistributedTracing:      false,
-				features.InitialCorruptCheck:     true,
-			},
-		},
-		{
-			name:                            "can set feature gate experimentalInitialCorruptCheck to false from experimental flag",
-			experimentalInitialCorruptCheck: "false",
-			expectedFeatures: map[featuregate.Feature]bool{
-				features.StopGRPCServiceOnDefrag: false,
-				features.DistributedTracing:      false,
-				features.InitialCorruptCheck:     false,
-			},
-		},
-		{
-			name:                   "can set feature gate StopGRPCServiceOnDefrag to true from feature gate flag",
-			serverFeatureGatesJSON: "StopGRPCServiceOnDefrag=true",
-			expectedFeatures: map[featuregate.Feature]bool{
-				features.StopGRPCServiceOnDefrag: true,
-				features.DistributedTracing:      false,
-				features.InitialCorruptCheck:     false,
-			},
+			name:                   "can set feature gate to false from feature gate flag",
+			serverFeatureGatesJSON: fgname + "=false",
+			expectedFeatures:       defaultFeatureGatesWithFeaturegate(fg, false),
 		},
+	}
+}
+
+type featureTestCase struct {
+	name                   string
+	serverFeatureGatesJSON string
+	expectErr              bool
+	expectedFeatures       map[featuregate.Feature]bool
+	experimentalKey        string
```

Comment:
```suggestion
	experimentalFlagName        string
```
or experimentalJSONField?